dask/bytes/http.py

```diff
@@ -110,7 +110,7 @@ def __init__(self, url, session=None, block_size=None, **kwargs):
         try:
             self.size = file_size(url, self.session, allow_redirects=True,
                                   **self.kwargs)
-        except ValueError:
+        except (ValueError, requests.HTTPError):
```

Comment:
The second exception is what a failed HEAD gives (methodnotallowed or some such)? Do you think it would be useful to give more verbose errors within `HTTPFile` for when things go wrong with naive usage?

Reply:
Yes, this is the error thrown by the failed `HEAD`. I can certainly try to give a more descriptive error here.